### PR TITLE
Configurable integration of KEB with LM and Reconciler

### PR DESCRIPTION
**Pull request by wozniakjan**
**Description**

KEB should be configurable to dynamically enable and disable its integration with Lifecycle Manager as well as Reconciler.

related to: https://github.com/kyma-project/kyma/issues/15445

<details>
<summary>kcp events</summary>

the lifecycle manager steps are highlighted

```diff
$ kcp rt --events -i test-jw256
GLOBALACCOUNT ID                       SUBACCOUNT ID                          SHOOT       REGION         PLAN   CREATED AT            STATE            KYMA VERSION
e449f875-b5b2-4485-b7c0-98725c0571bf   19ea2000-865d-42fd-8d6f-3e3e2dcce3c8   c-98ca31a   eu-central-1   aws    2023/01/04 09:59:12   deprovisioning   2.9.3
 ˫provision operation 4b7134a3-2dac-4e81-8ba2-596def96149a: succeeded
  ˫info   89m                         processing step: Starting
  ˫info   89m                         processing step: Provision_Initialization
  ˫info   89m                         processing step: Init_Kyma_Template
  ˫info   89m                         processing step: Resolve_Target_Secret
  ˫info   89m                         processing step: AVS_Create_Internal_Eval_Step
  ˫info   89m                         processing step: EDP_Registration
  ˫info   89m                         processing step: Overrides_From_Secrets_And_Config_Step
  ˫info   89m                         processing step: BTPOperatorOverrides
  ˫info   89m                         processing step: Create_Runtime_Without_Kyma
  ˫info   89m                         processing step: Check_Runtime
  ˫info   83m ago (4x in last 89m)    step Check_Runtime sleeping for 2m0s
  ˫info   81m                         processing step: Get_Kubeconfig
  ˫info   81m                         processing step: Inject_BTP_Operator_Credentials
+ ˫info   81m                         processing step: Sync_Kubeconfig
  ˫info   81m                         processing step: Create_Cluster_Configuration
  ˫info   81m                         processing step: Check_Cluster_Configuration
  ˫info   77m ago (10x in last 81m)   step Check_Cluster_Configuration sleeping for 30s
+ ˫info   76m                         processing step: Apply_Kyma
  ˫info   76m                         processing step: AVS_Create_External_Eval_Step
  ˫info   76m                         processing step: AVS_Tags
 ˫deprovision operation e5ebe9ef-6f2f-4806-9e1e-256ff8d7840a: in progress
  ˫info   2m17s                       processing step: Initialisation
  ˫info   2m17s                       processing step: BTPOperator_Cleanup
  ˫info   2m17s                       processing step: De-provision_AVS_Evaluations
  ˫info   2m12s                       processing step: EDP_Deregistration
+ ˫info   2m12s                       processing step: Delete_Kyma_Resource
+ ˫info   2m12s                       processing step: Check_Kyma_Resource_Deleted
  ˫info   2m12s                       step Check_Kyma_Resource_Deleted sleeping for 15s
  ˫info   117s                        processing step: Deregister_Cluster
  ˫info   117s                        processing step: Check_Cluster_Deregistration
  ˪info   26s ago (4x in last 117s)   step Check_Cluster_Deregistration sleeping for 30s
```

</details>

### CODE REVIEW

File: components/kyma-environment-broker/cmd/broker/main.go [hunk 1]
```diff
@@ -782,7 +788,7 @@ func NewUpdateProcessingQueue(ctx context.Context, manager *process.StagedManage
 		{
 			stage:     "btp-operator",
 			step:      update.NewApplyReconcilerConfigurationStep(db.Operations(), db.RuntimeStates(), reconcilerClient),
-			condition: update.RequiresReconcilerUpdate,
+			condition: requiresReconcilerUpdate,
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/cmd/broker/main.go:791]: why did you replace the value of the condition in line 757? you can leave the condition as it is and use `disabled` field together

Comment (wozniakjan) [reply components/kyma-environment-broker/cmd/broker/main.go:791]: the update queue doesn't have `disabled` afaict, only `condition`
https://github.com/kyma-project/control-plane/blob/bf84c20d9f0b9eca0f1e92afae23efe2cc4e2f8c/components/kyma-environment-broker/cmd/broker/main.go#L760-L764

**Review by piotrmiskiewicz — APPROVED** [review]

**Pull request merged**